ReSwift/Utils/TypeHelper.swift

```diff
@@ -9,14 +9,12 @@
 import Foundation
 
 /**
- Method is only used internally in Swift Flow to cast the generic `StateType` to a specific
- type expected by reducers / store subscribers.
-*/
-func withSpecificTypes<SpecificStateType, Action>(state: StateType,
-    action: Action, @noescape function: (state: SpecificStateType, action: Action)
-    -> SpecificStateType) -> StateType {
-
-    guard let specificStateType = state as? SpecificStateType else { return state }
+ Tries to cast a generic `StateType` to the specific type expected by reducers / store subscribers.
 
-    return function(state: specificStateType, action: action) as! StateType
-}
+ **NOTE: Method is only used internally.**
+*/
+func castToExpectedType<SpecificStateType, Action>(state: StateType, action: Action,
+	@noescape function: (action: Action, state: SpecificStateType)
+	-> SpecificStateType) -> SpecificStateType? {
+	return state as? SpecificStateType
+}
```

Comment:
Trailing Newline Violation: Files should have a single trailing newline. (trailing_newline)